### PR TITLE
Work/appimage updater dialog

### PR DESCRIPTION
**Pull request by erikjv**
Fixes: #11739

### CODE REVIEW

File: src/gui/updater/appimageupdater.cpp [hunk 1]
```diff
@@ -161,14 +161,15 @@ void AppImageUpdater::versionInfoArrived(const UpdateInfo &info)
         return;
     }
 
-    auto dialog = new AppImageUpdateAvailableDialog(currentVersion, newVersion, ocApp()->gui()->settingsDialog());
+    auto widget = new AppImageUpdateAvailableWidget(currentVersion, newVersion, ocApp()->gui()->settingsDialog());
+    widget->setAttribute(Qt::WA_DeleteOnClose);
```
Comment (TheOneRing) [inline src/gui/updater/appimageupdater.cpp:165]: I guess the close event will never be emitted as this is a plain widget.
I guess we should explicitly handle the delete in connects, `accepted`...?

Comment (erikjv) [reply src/gui/updater/appimageupdater.cpp:165]: That's done in the widget: the signal handler for both buttons calls a `close()` after the signal emission.

File: src/gui/updater/appimageupdater.cpp [hunk 2]
```diff
@@ -161,14 +161,24 @@ void AppImageUpdater::versionInfoArrived(const UpdateInfo &info)
...
 
-    auto dialog = new AppImageUpdateAvailableDialog(currentVersion, newVersion, ocApp()->gui()->settingsDialog());
+    QPointer<AppImageUpdateAvailableWidget> widget(new AppImageUpdateAvailableWidget(currentVersion, newVersion, ocApp()->gui()->settingsDialog()));
+    widget->setAttribute(Qt::WA_DeleteOnClose);
 
-    connect(dialog, &AppImageUpdateAvailableDialog::skipUpdateButtonClicked, this, [newVersion]() {
+    connect(widget, &AppImageUpdateAvailableWidget::skipUpdateButtonClicked, this, [newVersion, widget]() {
         qCInfo(lcUpdater) << "Update" << newVersion << "skipped by user";
         setPreviouslySkippedVersion(newVersion);
+        if (widget) {
```
Comment (TheOneRing) [inline src/gui/updater/appimageupdater.cpp:170]: As the connect is on the widget it must still exist at this time.
Please also use deleteLater.

File: src/gui/updater/appimageupdater.cpp (hunk 1 above)
Comment (TheOneRing) [reply src/gui/updater/appimageupdater.cpp:165]: Please fix

File: src/gui/updater/appimageupdater.cpp [hunk 3]
```diff
@@ -182,10 +192,14 @@ void AppImageUpdater::versionInfoArrived(const UpdateInfo &info)
 
         setDownloadState(Downloading);
         appImageUpdaterShim->startUpdateInBackground();
+
+        if (widget) {
```
Comment (TheOneRing) [inline src/gui/updater/appimageupdater.cpp:196]: see above

**Review by TheOneRing — CHANGES_REQUESTED** [review]

**Review by TheOneRing — APPROVED** [review]

**Pull request merged**